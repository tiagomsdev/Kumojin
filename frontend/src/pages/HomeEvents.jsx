import {useState} from "react";
import {
  AppBar,
  Toolbar,
  Typography,
  IconButton,
  Grid
} from "@material-ui/core";
import MenuIcon from "@mui/icons-material/Menu";
import EventDialog from "../components/CreateEvent/EventDialog";
import EventList from "../components/EventList/EventList";
import {getUserTimezone, getUserDateFromUtc} from "../helpers/momentTimeZone";


function HomeEvents() {
   const[eventList, setEventList] = useState([]);

   const handleEventList = (data) =>{
       const newEventList  = [...data];
       setEventList(...eventList,handleEventTimeZone(newEventList));
   }

   function handleEventTimeZone(data){
      data.map((item) => (
        item.startDate = getUserDateFromUtc(item.startDate, getUserTimezone()),
        item.endDate = getUserDateFromUtc(item.endDate, getUserTimezone())
      ));
      return data;
   }

   const handleCreateEvent = (data) => {
      setEventList(oldEventList => [...oldEventList,...handleEventTimeZone(data)]);
   }

  return (
    <Grid container spacing={8.4}  justifyContent="left">
      <Grid item md={12}>
          <AppBar data-testid="homeEvents-appBar">
            <Toolbar>
              <IconButton
                size="large"
                edge="start"
                color="inherit"
                aria-label="menu"
                sx={{ mr: 2 }}
              >
                <MenuIcon />
              </IconButton>
              <Typography variant="h6" 
            noWrap
            component="div"
            sx={{ padding:2 }}>
                Liste d'événements
              </Typography>
              <EventDialog handleCreateEvent={handleCreateEvent}/>
            </Toolbar>
          </AppBar>    
      </Grid>
      <Grid item justifyContent="left">
          <EventList handleEventList={handleEventList} eventList={eventList}/>
      </Grid>
    </Grid>
  );
}


export default HomeEvents;
