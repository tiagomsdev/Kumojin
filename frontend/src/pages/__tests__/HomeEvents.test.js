import {render, screen, cleanup} from '@testing-library/react';
import HomeEvents from "../HomeEvents";
import renderer from "react-test-renderer";

test('should render HomeEvents component', () => {
    
    render(<HomeEvents/>);

    const appBarElement = screen.getByTestId('homeEvents-appBar');


    expect(appBarElement).toBeInTheDocument();
    expect(appBarElement).toHaveTextContent("Liste d'événements");
    
});

test('should matches HomeEvents snapshot', () =>{
    
    const homeEventsTree = renderer.create(<HomeEvents/>).toJSON();
    expect(homeEventsTree).toMatchSnapshot();
});