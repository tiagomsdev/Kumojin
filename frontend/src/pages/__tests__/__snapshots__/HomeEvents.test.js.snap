// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matches HomeEvents snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8.4 css-ljhyho-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12 css-15fi2ac-MuiGrid-root"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      data-testid="homeEvents-appBar"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
        <div
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mrw4xf-MuiTypography-root"
        >
          Liste d'événements
        </div>
        <div>
          <button
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="newEvent-button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            + Nouveau
          </button>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-hb9o7r-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="simple table"
        className="MuiTable-root css-nv4u2v-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          data-testid="eventList-tableHead"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Nom de l'événement
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Description de l'événement
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Date/Heure de début
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Date/Heure de fin
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          data-testid="eventList-tablebody"
          role={null}
        />
      </table>
    </div>
  </div>
</div>
`;
